homeassistant/components/lcn/helpers.py

```diff
@@ -476,6 +481,25 @@ def is_address(value: str) -> tuple[AddressType, str]:
     raise ValueError(f"{value} is not a valid address string")
 
 
+def address_to_device_id(
+    hass: HomeAssistant, address: AddressType, host_name: str | None = None
+) -> str | None:
+    """Convert LCN address to device_id."""
+    for config_entry in hass.config_entries.async_entries(DOMAIN):
+        if (host_name is None) or (config_entry.title == host_name):
+            break
+    else:
+        return None
```

Comment:
what does this do?

Reply:
The host_name is basically the title/name of the integration entry and known to the user (because he can see/edit this name in the UI).
For LCN we have a config_entry/integration entry for each host (or let's say hardware hub). The address identifies the hardware device (module) connected to that specific host.
A LCN hardware device is identified by the host_name and the hardware address (two numeric IDs which might be the same for two separate hosts).

The iteration does the following:
- if the host name is not provided (is None), take the first config_entry
- otherwise check if there is a config_entry.title with the given host_name
- if both checks fail, there is either no config_entry with matching title or no config_entry at all -> return None

Reply:
the thing is, config entry title is considered userland, so they can change the title to whatever they want. And that makes me hesitant to make logic against.

Reply:
I know. The thing is, I don't see any other satisfactory way for the user to identify the host device (and thus the integration entry). There is no *name* or *identifier* the hardware provides. The only possibility would be the IP address. But I don't find that satisfactory either.

Reply:
Would it be possible to use the config entry id instead?

Reply:
I guess so. But the config entry id is just as uncomfortable to access as the device_id. The `address_to_device_id` action's intention is to provide a "easy" way to get the device_id with parameters the users actually knows.

However, if it is not acceptable to “abuse” the config_entry.title, I would suggest to remove the “address_to_device_id” action completely and leave it to the user to somehow get the device_id for the scripted automations. (It doesn't matter for the UI, as there are appropriate selectors there.)

Reply:
I don't want to defend my approach at all costs. :smile:
For me, changing the config_entry.title is comparable to changing the entity_id. I could also change the latter from the UI and it would kill my automations.

Reply:
Don't worry, I'm just trying to push back with some reasons why we should not do it to discover more on why this is a good idea.

But in that case you wrote a YAML automation with a direct link to the entity id. If you used the entity entry id (the one that is used in the entity registry) (you would automatically use this if you create automations in the UI), it would still work. Your example would be someone pointing at a box, and then moving the box and then complaining that the box isn't there anymore. To stay in this comparison, since we consider entry title userland, we try to enable users to move that box wherever they like (it would not surprise me if there were still remnants in the code where this isn't applied yet, but all the new code certainly is). So if the users expect to move that box without consequences (which is true for let's say 90+%), we should try to maintain that expectation.

The thing with the title is, people change it. They find out it doesn't work. They want to reset it, forget their old name. They have to remove the integration and add it again. I think that's how the average user would tackle it, and if we can avoid that, I'd prefer that :)

Reply:
Ok, if I can't rely on the user not understanding that if they move the box, they won't find it in the previous location (even though it's in the docs!), I'm also in favor of taking away the option. 🤣 
Okay, let's keep it short. I can't think of any other way except the IP/port address or the config_entry.entry_id.  None of which I find really user-friendly and reliable.
Let's remove the `address_to_device_id` action. I don't think it should be a big deal for anyone to stick to the device_id if they really need to write automations as yaml.

Reply:
But what do we use this function for again? (Im currently on mobile). Because if this is used to reference a config entry, we could also use a config entry selector

Reply:
Thankfully, I've already written the docs:  https://github.com/home-assistant/home-assistant.io/pull/35600/files#:~:text=When%20actions%20are,%7B%25%20endraw%20%25%7D

The most important point is when using the actions in an automation which is written as yaml (you don't have selectors there).
The last example in the linked text is important and the use case for the `address_to_device_id` action.

Reply:
Right, but this sounds to me that it could be replaced with a config entry selector. Which would be more elegant. And yes, that is not the user friendliest in YAML, but that's only because there is no nice way to get the config entry id. But going to the dev tools and trying to execute the action and then going to YAML mode would at least get you the entry id used and then its just another value you copy and paste. WDYT about that?

Reply:
The device selector is actually sufficient, as we only need the device_id. Querying the device_id via the developer tools was the method that I personally did not find very user-friendly.  But as I said, I don't have a better solution, and we shouldn't make an exception just for this integration.

Reply:
Maybe this is a good WTH idea. Have a way to go to a device in the device overview and copy the id. It would probably make this better

Reply:
Thank you for your endless patience! I removed the `address_to_device_id` action and also adapted the docs PR.

Reply:
Ah, one additional comment: Unfortunately we have to keep the `address_to_device_id` method in `helpers.py` (the one that contains the code you highlighted here).
But this is only used for the deprecated `address` parameter and will vanish as soon as the `address` parameter has been removed.

Reply:
To push this forward, I reverted the code back to the old state and only added the logic for the device_id parameter. So, no `config_entry.title` anymore.